src/entities/entity.h

```diff
@@ -0,0 +1,67 @@
+#ifndef ENTITY_H
+#define ENTITY_H
+
+#include <QObject>
+#include <QThread>
+#include <QMutex>
+
+#include <src/utils/timer/timer.h>
+
+enum EntityType {
```

Comment:
These ids are used to enum the used modules, they are required in the constructors.